### ISSUE TITLE
Search Window, Correlation Window, and Node Spacing

### ISSUE THREAD

**Issue by Alireza8999** [ElsevierSoftwareX/SOFTX-D-17-00061#15]
I have figured out how to run the program, however, the calculations for search window, correlation window, and node spacing seem to be off. I was wondering how exactly these are calculated. My images are tiff format and 1000x1000 pixels. I only need a subset of the image read for analysis. 

Below are my current running parameters:
search_window = [[-20, 3], [-20, 2], [-20, 3]]
correlation_window = [3, 3, 3]
node_spacing = [5, 5, 5]

and all that happens is that in the tsv file the values are -2 to -12,-20, -20 for the displacement values. 

Thank you!

**Comment by erikatudisco** [comment]
Hej,

The correlation window and search window depends on your images and applied deformation respectively and it is very difficult to give an advise without seeing the images and/or knowing the experiment you run. The node spacing gives you basically the number or pixels in the output image. You normally start with a large node spacing and then when the correlation works you can decrease it to have nicer images.

The fact that  in the output all the x and y values are -20 it's and indication that the search window is too small and therefore not good correlation is found. The first I would try is to use a large symmetrical search window (with a big node spacing). the correlation window could also be too small but this really depends on your images

Hope this helps!

Erika Tudisco
Associate Professor

Lund University
Faculty of Engineering LTH
Dept. of Construction Sciences
Geotechnical Engineering
Box 118, SE 221 00 Lund
Visiting address: John Ericssons väg 1
Telephone: +46 46 22 20 356
https://www.geoteknik.lth.se/english/staff/erika-tudisco/


On Sun, 2022-11-06 at 13:13 -0800, Alireza8999 wrote:

I have figured out how to run the program, however, the calculations for search window, correlation window, and node spacing seem to be off. I was wondering how exactly these are calculated. My images are tiff format and 1000x1000 pixels. I only need a subset of the image read for analysis.

Below are my current running parameters:
search_window = [[-20, 3], [-20, 2], [-20, 3]]
correlation_window = [3, 3, 3]
node_spacing = [5, 5, 5]

and all that happens is that in the tsv file the values are -2 to -12,-20, -20 for the displacement values.

Thank you!

**Comment by Alireza8999** [comment]
We ran the program using the following input parameters. 
```
Advanced = 1
DIR_image1 = './test/0'
DIR_image2 = './test/12'
DIR_out = './test/results/3d'

search_window = [[-3, 3], [-30, 30], [-30, 30]]
correlation_window = [1, 1, 1]
node_spacing = [10, 10, 10]

image_format = 'TIFF'

grey_low_threshold = 0.3
output_name = 'DIC_advanced'

kinematics_median_filter = 0
```
And our images are coming out with all kinds of colors when we do plot them. I ran your images using the same parameters and the colors are clean. I was wondering if was potentially any filters you ran your images through prior to running the program or perhaps my data is just dirty. I have attached the images of my runs. Capture 4 (second image) is my run and Capture 6 (first image) is your images

![Capture6](https://user-images.githubusercontent.com/117602153/203410437-a8d5eab6-bddd-4f06-b159-4386130fc95f.JPG)

![Capture4](https://user-images.githubusercontent.com/117602153/203410385-c060179b-4b54-4791-aa11-06f4d40b3e3f.JPG)

I appreciate you answering earlier, it helped a lot. 
Thank you!
